proposesolution/ai_test.go

```diff
@@ -0,0 +1,30 @@
+package proposesolution_test
+
+import (
+	"testing"
+
+	"github.com/securego/gosec/v2/issue"
+	"github.com/securego/gosec/v2/proposesolution"
+)
+
+func TestGenerateSolution(t *testing.T) {
+	aiApiProvider := proposesolution.GeminiProvider
+	aiApiKey := "test-api-key" // Replace with a valid API key for actual testing
+
+	issues := []*issue.Issue{
+		{
+			What: "Blocklisted import crypto/md5: weak cryptographic primitive",
+		},
+	}
+
+	err := proposesolution.GenerateSolution(aiApiProvider, aiApiKey, issues)
```

Comment:
This is calling the real gemeni API which is not acceptable for a unit test. Inside of unit tests we don't call external APIs. 

We need to mock the genai.NewClient if you want to add a unit test.